app/estimate_square_size.go

```diff
@@ -28,14 +40,14 @@ func estimateSquareSize(txs []parsedTx) (squareSize uint64, nonreserveStart int)
 		blobSharesUsed += blobtypes.BlobTxSharesUsed(ptx.blobTx)
 	}
 
-	// assume that we have to add a lot of padding by simply doubling the number
-	// of shares used
-	//
-	// TODO: use a more precise estimation that doesn't over
-	// estimate as much
-	totalSharesUsed := uint64(txSharesUsed + blobSharesUsed)
-	totalSharesUsed *= 2
-	minSize := uint64(math.Sqrt(float64(totalSharesUsed)))
+	totalSharesUsed := float64(txSharesUsed + blobSharesUsed)
+	if totalSharesUsed <= 1 {
+		return appconsts.DefaultMinSquareSize, txSharesUsed
+	}
+	// increase the total shares used by the worst case padding ratio
+	totalSharesUsed *= worstCasePaddingCoefficient
```

Comment:
Because we intorduced the `worstCasePaddingBase` we don't need to multiply `totalSharesUsed` by the coefficient. Only blobSharesUsed.